Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/ComponentViewController.swift

```diff
@@ -181,7 +169,7 @@ extension ComponentsViewController: UITableViewDelegate, UITableViewDataSource {
                 tableView.rowHeight = 100
                 number.numberFieldPlacholder = "0"
                 number.titleText = componentHeaderText[i]
-                number.currentPage = Int(numberFieldString ?? 0)
+                number.currentPage = numberFieldString ?? 0
                 number.numberField.text = "\(numberFieldString ?? 0)"
                 number.frame = CGRect(x: 10, y: 10, width: tableView.bounds.width - 20, height: tableView.rowHeight)
```

Comment:
Wouldn't you want things like the rect setup to be within the component of Signature and not set as your mapping it out?

Ideally you would just detect which field is on this Document, then map out each one with one line of code instantiating the component and then passing data into them. It looks like you are grabbing an instance of each one and creating its UI on the fly?

I may be confused on what this is doing so let me know

Reply:
I'm doing exactly the same as you thought like to detect which component  is on the document and then map each one with the component and passing the value/data to each component coming from the API